### ISSUE TITLE
Catboost for semi supervised learning possible?

### ISSUE THREAD

**Issue by thomas9000** [catboost/catboost#525]
Hi
Can Catboost support semi supervised learning considering many organizations does not have a lot of full training data with label output (Y) > 100,000 ?

**Comment by annaveronika** [comment]
No, unfortunately we don't support semi supervised learning, and we don't plan to add this support in the close future.

**Comment by MohitBurkule** [comment]
You can do it with sklearn  [SelfTrainer](https://scikit-learn.org/stable/modules/generated/sklearn.semi_supervised.SelfTrainingClassifier.html#sklearn.semi_supervised.SelfTrainingClassifier)

code would look something like 

```
from sklearn.semi_supervised import SelfTrainingClassifier

cb=CatBoostClassifier(task_type="CPU",depth=3) # any params 
self_training_model = SelfTrainingClassifier(cb,threshold=0.5)
self_training_model.fit(X_semilabelled,y_semilabelled)
self_training_model.predict(X_test)
```


one disadvantage of this method is that since sklearn internally calls base model (catboost) fit with only 2 paramters X and y, you cannot pass additional parameters to cb.fit() , hence i made a  [fork of sklearn at my github](https://github.com/MohitBurkule/scikit-learn)  which allows you to pass parameters to self_training_model.fit which would be passed on to the cb.fit internally 

so you could do
```
from sklearn.semi_supervised import SelfTrainingClassifier

cb=CatBoostClassifier(task_type="CPU",depth=3) # any params 
self_training_model = SelfTrainingClassifier(cb,threshold=0.5)
self_training_model.fit(X_semilabelled,y_semilabelled,use_best_model=True,eval_set=(x_val,y_val1),cat_features=[0])
self_training_model.predict(X_test)
```
